xla/service/gpu/transforms/collective_permute_cycle_decomposer_test.cc

```diff
@@ -187,6 +206,76 @@ TEST_F(CollectivePermuteCycleDecomposerTest, ForwardCycleWithMatmul) {
               HasSubstr("_xla_send_recv_validation={{0,7},{1,8},{2,9}}"));
 }
 
+TEST_F(CollectivePermuteCycleDecomposerTest, ForwardCycleWithMatmulNoChannel) {
+  // This test is a variation of the test above, that also checks that the
+  // collectives have no channel-id across replicas, and that the collectives
+  // are combined based on the value of replica-id.
+  const absl::string_view kModuleStr = R"(
+  HloModule test
+
+  while_cond {
+    param = (u32[], f32[2,2], f32[2,2]) parameter(0)
+    iter = u32[] get-tuple-element(param), index=0
+    max_iter = u32[] constant(3)
+    ROOT cmp = pred[] compare(iter, max_iter), direction=LT
+  }
+
+  while_body {
+    param = (u32[], f32[2,2], f32[2,2]) parameter(0)
+    iter = u32[] get-tuple-element(param), index=0
+    data = f32[2,2] get-tuple-element(param), index=1
+    weights = f32[2,2] get-tuple-element(param), index=2
+    cp = f32[2,2] collective-permute(data),  
+      source_target_pairs={{0,1}, {1,2}, {2,3}, {3,0}},
+      frontend_attributes={_xla_send_recv_validation="{{0,7},{1,8},{2,9},{3,10}}"}
+    matmul = f32[2,2] dot(weights, cp), lhs_contracting_dims={1}, rhs_contracting_dims={0}
+    iter_increment = u32[] constant(1)
+    next_iter = u32[] add(iter, iter_increment)
+    ROOT result = (u32[], f32[2,2], f32[2,2]) tuple(next_iter, matmul, weights)
+  }
+
+  ENTRY test_computation {
+    iter = u32[] constant(0)
+    data = f32[2,2] parameter(0)
+    weights = f32[2,2] parameter(1)
+    input = (u32[], f32[2,2], f32[2,2]) tuple(iter, data, weights)
+    while_res = (u32[], f32[2,2], f32[2,2]) while(input), condition=while_cond, body=while_body
+    ROOT data_out = f32[2,2] get-tuple-element(while_res), index=1
+  }
+  )";
+  TF_ASSERT_OK_AND_ASSIGN(std::unique_ptr<HloModule> module,
+                          ParseAndReturnUnverifiedModule((kModuleStr)));
+  CollectivePermuteCycleDecomposer decomposer(/*threshold_in_bytes=*/0);
+  TF_ASSERT_OK_AND_ASSIGN(bool changed, decomposer.Run(module.get()));
+  EXPECT_TRUE(changed);
+  HloPrintOptions options;
+  options.set_print_operand_shape(false);
+  TF_CHECK_OK(VerifyHloModule(module.get(), false, true));
+  EXPECT_TRUE(*RunFileCheck(module->ToString(options), R"(
+    CHECK: %while_body {{.+}} {
```

Comment:
nit: indentation

Reply:
I don't understand this comment: would you like for it to be of the form `// CHECK:` instead? I have updated it to this form, but the indentation itself looks fine to me, I think. Can you point out specifically how you want this changed?